### PR TITLE
fix(a11y): Fixing lost focus when task modal is closed

### PR DESCRIPTION
**Pull request by jeeyi**

### CODE REVIEW

File: src/elements/content-sidebar/AddTaskButton.js [hunk 1]
```diff
@@ -44,18 +46,29 @@ class AddTaskButton extends React.Component<Props, State> {
...
         this.setState({ isTaskFormOpen: false, error: null });
+        if (this.buttonRef) {
+            this.buttonRef.focus();
+        }
         onTaskModalClose();
     };
 
     handleSubmitError = (e: ElementsXhrError) => this.setState({ error: e });
 
+    setAddTaskButtonRef = (ref: { current: null | HTMLButtonElement, focus: Function }) => {
```
Comment (booc0mtaco) [inline src/elements/content-sidebar/AddTaskButton.js]: @alexkrolick thoughts on this bit?

Comment (alexkrolick) [reply src/elements/content-sidebar/AddTaskButton.js]: shouldn't focus be callable on the HTMLButtonElement? `this.buttonRef.current.focus()`

Comment (alexkrolick) [reply src/elements/content-sidebar/AddTaskButton.js]: Looks like we are kind of mixing React.createRef with the function pattern here - should use one or the other.

File: src/elements/content-sidebar/AddTaskButton.js [hunk 2]
```diff
@@ -22,6 +22,8 @@ type State = {
 };
 
 class AddTaskButton extends React.Component<Props, State> {
+    buttonRef: { current: null | HTMLButtonElement, focus: Function };
```
Comment (alexkrolick) [inline src/elements/content-sidebar/AddTaskButton.js]:
```suggestion
    buttonRef: React.Ref<HTMLButtonElement>; // same as { current: ?HTMLButtonElement }
```

File: src/elements/content-sidebar/AddTaskButton.js [hunk 3]
```diff
@@ -44,18 +46,29 @@ class AddTaskButton extends React.Component<Props, State> {
     handleModalClose = () => {
         const { onTaskModalClose } = this.props;
         this.setState({ isTaskFormOpen: false, error: null });
+        if (this.buttonRef) {
+            this.buttonRef.focus();
+        }
```
Comment (alexkrolick) [inline src/elements/content-sidebar/AddTaskButton.js]:
```suggestion
        if (this.buttonRef.current) {
            this.buttonRef.current.focus();
        }
```

File: src/elements/content-sidebar/AddTaskButton.js [hunk 4]
```diff
@@ -44,18 +46,29 @@ class AddTaskButton extends React.Component<Props, State> {
...
         onTaskModalClose();
     };
 
     handleSubmitError = (e: ElementsXhrError) => this.setState({ error: e });
 
+    setAddTaskButtonRef = (ref: { current: null | HTMLButtonElement, focus: Function }) => {
+        this.buttonRef = ref;
+    };
```
Comment (alexkrolick) [inline src/elements/content-sidebar/AddTaskButton.js]:
```suggestion
    setAddTaskButtonRef = (element: ?HTMLButtonElement) => {
        this.buttonRef.current = element;
    };
```

File: src/elements/content-sidebar/AddTaskButton.js (hunk 1 above)
Comment (jeeyi) [reply src/elements/content-sidebar/AddTaskButton.js]: @alexkrolick Updated to use React.createRef.

File: src/elements/content-sidebar/AddTaskButton.js (hunk 3 above)
Comment (jeeyi) [reply src/elements/content-sidebar/AddTaskButton.js]: Done

File: src/elements/content-sidebar/AddTaskMenu.js [hunk 5]
```diff
@@ -17,11 +17,12 @@ import './AddTaskMenu.scss';
 type Props = {
     isDisabled: boolean,
     onMenuItemClick: (taskType: TaskType) => void,
+    setAddTaskButtonRef?: Function,
```
Comment (booc0mtaco) [inline src/elements/content-sidebar/AddTaskMenu.js]: we can type this more strictly, i.e., use `(element: HTMLButtonElement) => void` or similar

Comment (jeeyi) [reply src/elements/content-sidebar/AddTaskMenu.js]: Yes

File: src/elements/content-sidebar/AddTaskMenu.js [hunk 6]
```diff
@@ -17,11 +17,12 @@ import './AddTaskMenu.scss';
 type Props = {
     isDisabled: boolean,
     onMenuItemClick: (taskType: TaskType) => void,
+    setAddTaskButtonRef?: (element: HTMLButtonElement) => void,
 };
 
 const AddTaskMenu = (props: Props) => (
     <DropdownMenu constrainToScrollParent isRightAligned>
-        <Button isDisabled={props.isDisabled} type="button">
+        <Button isDisabled={props.isDisabled} type="button" setRef={props.setAddTaskButtonRef}>
```
Comment (mxiao6) [inline src/elements/content-sidebar/AddTaskMenu.js]: You can destruct the `props` in function parameter to extract `isDisabled` and `setAddTaskButtonRef`.

Comment (jstoffan) [reply src/elements/content-sidebar/AddTaskMenu.js]: We can and should, but it looks like the whole file is written this way. Maybe as a cleanup PR?

Comment (booc0mtaco) [reply src/elements/content-sidebar/AddTaskMenu.js]: What's our rationale behind enforcing destructuring?

Comment (jstoffan) [reply src/elements/content-sidebar/AddTaskMenu.js]: It's defined in our eslint config, I think from airbnb's, as [react/destructuring-assignment](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/destructuring-assignment.md). This project it has it disabled due to there being so many instances of noncompliance..

**Pull request merged**